src/System.Windows.Forms.Primitives/src/System/Windows/Forms/SinglyLinkedList.cs

```diff
@@ -56,13 +68,116 @@ public Node AddLast(T value)
 
     public Enumerator GetEnumerator() => new(this);
 
-    public class Node
+    public void Clear()
+    {
+        Node? current = First;
+        while (current is not null)
+        {
+            Node temp = current;
+            current = current.Next;
+            temp.Invalidate();
+        }
+
+        First = null;
+        Last = null;
+        Count = 0;
+    }
+
+    public bool Contains(T value)
+    {
+        return Find(value) is not null;
+    }
+
+    public Node? Find(T value)
+    {
+        Node? node = First;
+        EqualityComparer<T> c = EqualityComparer<T>.Default;
+        if (node is not null)
+        {
+            if (value is not null)
+            {
+                while (node is not null)
+                {
+                    if (c.Equals(node!.Value, value))
+                    {
+                        return node;
+                    }
+
+                    node = node.Next;
+                }
+            }
+            else
+            {
+                while (node is not null)
+                {
+                    if (node!.Value is null)
```

Comment:
```suggestion
                    if (node.Value is null)
```

Also, in this loop and the one above, you could put this `if` into the `while`, and return `node` at the end of the method, but that's more of a style preference